packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -491,11 +506,25 @@ export class GadgetConnection {
     });
   }
 
-  private requestHeaders() {
+  private async requestHeaders() {
     const headers: Record<string, string> = {};
 
-    if (this.authenticationMode == AuthenticationMode.InternalAuthToken) {
-      headers.authorization = "Basic " + base64("gadget-internal" + ":" + this.options.authenticationMode!.internalAuthToken!);
+    if (this.authenticationMode == AuthenticationMode.Internal || this.authenticationMode == AuthenticationMode.InternalAuthToken) {
+      const authToken =
+        this.authenticationMode == AuthenticationMode.Internal
+          ? this.options.authenticationMode!.internal!.authToken
+          : this.options.authenticationMode!.internalAuthToken!;
+
+      headers.authorization = "Basic " + base64("gadget-internal" + ":" + authToken);
+
+      if (this.authenticationMode == AuthenticationMode.Internal && this.options.authenticationMode!.internal!.useSession) {
+        headers["x-gadget-use-internal-session"] = "true";
+
+        const sessionId = await this.options.authenticationMode!.internal!.getSessionId?.();
```

Comment:
Tiny nit: what do you think about calling this `masquerade` or `impersonate` or something like that? I feel like it might be a little clearer to call the header that and the function that for the next person who comes along

Reply:
`x-gadget-act-as-internal-session`?